src/main/java/seedu/exercise/model/property/PropertyBook.java

```diff
@@ -68,28 +69,38 @@ public boolean isFullNameUsed(String fullName) {
     }
 
     /**
-     * Adds the prefix of a newly defined custom property.
+     * Adds the newly defined custom property into the manager.
      */
-    private void addPrefix(Prefix prefix) {
-        prefixes.add(prefix);
+    public void addCustomProperty(CustomProperty customProperty) {
+        Prefix newPrefix = customProperty.getPrefix();
+        String newFullName = customProperty.getFullName();
+        addPrefix(newPrefix);
+        addFullName(newFullName);
+        CUSTOM_PROPERTIES.add(customProperty);
+        updatePropertyPrefixes();
     }
 
     /**
-     * Adds the full name of a newly defined custom property.
+     * Removes the custom property with the associated {@code fullName}, if such a custom property
+     * exists.
      */
-    private void addFullName(String fullName) {
-        fullNames.add(fullName);
+    public void removeCustomProperty(String fullName) {
```

Comment:
Maybe can call it removeCustomPropertyByName to prevent confusion with the method below of the same name

Reply:
Or you can call the one below `removeCustomPropertyFromModel`

Reply:
The method below is actually a private method so it won't be able to be called by anything outside the class.